Sources/Apollo/RecordSet.swift

```diff
@@ -10,8 +10,17 @@ public struct RecordSet {
     storage[record.key] = record
   }
 
-  public mutating func clear() {
-    storage.removeAll()
+  public mutating func clear(_ clearingPolicy: CacheClearingPolicy = .allRecords) {
+    switch clearingPolicy._value {
+    case let .first(count): self.storage = .init(self.storage.dropFirst(count))
+
+    case let .last(count): self.storage = .init(self.storage.dropLast(count))
+
+    case let .allMatchingKeyPattern(pattern): self.storage = self.storage.filter { !$0.key.contains(pattern) }
```

Comment:
Does `contains` automatically handle globs?

Reply:
🤦 No, this is my bad. I'll need to remove the globs and then this will work.